general/mem_manager.hpp

```diff
@@ -31,16 +31,23 @@ enum class MemoryType
    HOST_32,        ///< Host memory; aligned at 32 bytes
    HOST_64,        ///< Host memory; aligned at 64 bytes
    HOST_DEBUG,     ///< Host memory; allocated from a "host-debug" pool
-   HOST_UMPIRE,    ///< Host memory; using Umpire
+   HOST_UMPIRE,    /**< Host memory; using an Umpire allocator which can be set
+                        with MemoryManager::SetUmpireHostAllocatorId */
    HOST_PINNED,    ///< Host memory: pinned (page-locked)
    MANAGED,        /**< Managed memory; using CUDA or HIP *MallocManaged
                         and *Free */
    DEVICE,         ///< Device memory; using CUDA or HIP *Malloc and *Free
    DEVICE_DEBUG,   /**< Pseudo-device memory; allocated on host from a
                         "device-debug" pool */
-   DEVICE_UMPIRE,  ///< Device memory; using Umpire
+   DEVICE_UMPIRE,  /**< Device memory; using an Umpire allocator which can be
+                        set with MemoryManager::SetUmpireDeviceAllocatorId */
+   DEVICE_UMPIRE_2, /**< Device memory; using a second Umpire allocator settable
```

Comment:
Yea

Reply:
Can that be reflected in the name, maybe `DEVICE_UMPIRE_TMP` instead of `DEVICE_UMPIRE_2`?

Reply:
We used to call it `DEVICE_TEMP_UMPIRE` but any one of these is fine

https://github.com/mfem/mfem/commit/802f21387316fb590dba85ca0e8e8bf8d19da2e4#diff-8c7013577433cbad6586f7299f99abea50089e068ea39a35bdcfb6f8f75f9214R79

Reply:
`DEVICE_TEMP_UMPIRE` is also good for me

Reply:
The idea is that this type just denotes a second Umpire device allocator -- how it is used is up to the application. That's why I want to avoid calling it `TMP` or `TEMP`.